Framework/API/src/ComponentInfo.cpp

```diff
@@ -13,16 +13,39 @@ namespace API {
 /**
  * Constructor
  * @param componentInfo : Beamline wrapped ComponentInfo
- * @param componentIds : Component Ids ordered by component index
+ * @param componentIds : l-value reference to ComponentIDs ordered by component
+ * index
  */
-ComponentInfo::ComponentInfo(const Beamline::ComponentInfo &componentInfo,
-                             std::vector<Geometry::ComponentID> &&componentIds)
+ComponentInfo::ComponentInfo(
+    const Mantid::Beamline::ComponentInfo &componentInfo,
+    std::vector<Mantid::Geometry::IComponent *> &componentIds)
     : m_componentInfo(componentInfo),
       m_componentIds(
           boost::make_shared<std::vector<Geometry::ComponentID>>(componentIds)),
       m_compIDToIndex(boost::make_shared<
           std::unordered_map<Geometry::IComponent *, size_t>>()) {
+  init();
+}
 
+/**
+ * Constructor
+ * @brief ComponentInfo::ComponentInfo
+ * @param componentInfo
+ * @param componentIds : r-value reference to ComponentIDs ordered by component
+ * index
+ */
+ComponentInfo::ComponentInfo(
+    const Mantid::Beamline::ComponentInfo &componentInfo,
+    std::vector<Mantid::Geometry::IComponent *> &&componentIds)
+    : m_componentInfo(componentInfo),
```

Comment:
Inside the r-value reference constructor `componentIds` is now an lvalue which should be able to bind to a non-const lvalue reference I think?

This seems to work, for example: 

```
struct Foo {
  Foo(std::vector<int> & ids) : cids(ids) {
     ids = {1,2,3,4,5};
  }
  Foo(std::vector<int> && ids) : Foo(ids) {} 
  std::vector<int> cids;
};

int main() {
    std::vector<int> my_ids;
    {
     Foo f(my_ids);
    }
    std::copy(std::begin(my_ids), std::end(my_ids),
              std::ostream_iterator<int>(std::cout, ","));   
}
```

Demo on [wandbox](https://wandbox.org/permlink/GMO6FlH7vVBpM19R).